action/protocol/rewarding/reward.go

```diff
@@ -94,8 +94,12 @@ func (p *Protocol) GrantBlockReward(
 	}
 	// If reward address doesn't exist, do nothing
 	if rewardAddrStr == "" {
-		log.S().Debugf("Producer %s doesn't have a reward address", producerAddrStr)
-		return nil, nil
+		if p.scheme == _standaloneScheme {
+			rewardAddrStr = producerAddrStr
```

Comment:
this is an `else` for the `if` in line 83, that `pp` is nil. Or an easier change is to have a poll protocol with only one candidate, who is the producer.